File: memento/exercise/Canvas.java
package memento.exercise;

import memento.TextEditorState;

public class Canvas {
    private String content;
    private String color;
    private String border;

    public Canvas() {
        this.content = "";
        this.color = "";
        this.border = "";
    }

    public CanvasState save() {
        return new CanvasState(content, color, border);
    }

    public void restore(CanvasState canvasState) {
        content = canvasState.getContent();
        color = canvasState.getColor();
        border = canvasState.getBorder();
    }

    public String getContent() {
        return content;
    }

    public String getColor() {
        return color;
    }

    public String getBorder() {
        return border;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public void setBorder(String border) {
        this.border = border;
    }
}

File: PHP/src/Behavioral/Memento/Exercise/Canvas.php
<?php

namespace App\Behavioral\Memento\Exercise;

class Canvas
{
    private $content;
    private $color;
    private $border;

    public function __construct()
    {
        $this->content = '';
        $this->color = '';
        $this->border = '';
    }

    /**
     * @return string
     */
    public function getContent(): string
    {
        return $this->content;
    }

    /**
     * Set the value of content
     *
     * @return  void
     */
    public function setContent($content)
    {
        $this->content = $content;

    }

    /**
     * @return string
     */
    public function getColor(): string
    {
        return $this->color;
    }

    /**
     * Set the value of color
     *
     * @return  void
     */
    public function setColor($color)
    {
        $this->color = $color;

    }

    /**
     * @return string
     */
    public function getBorder(): string
    {
        return $this->border;
    }

    /**
     * Set the value of border
     *
     * @return  void
     */
    public function setBorder($border)
    {
        $this->border = $border;

    }

    public function save(): CanvasState
    {
        return new CanvasState($this->content, $this->color, $this->border);
    }

    public function restore(CanvasState $canvasState): void
    {
        $this->content = $canvasState->getContent();
        $this->color = $canvasState->getColor();
        $this->border = $canvasState->getBorder();
    }
}
